src/cmd/manage_nsfs.js

```diff
@@ -334,93 +382,155 @@ async function fetch_account_data(argv, config_root, from_file) {
             }
         }, _.isUndefined);
     }
+
     data.creation_date = new Date().toISOString();
-    if (action === 'update') {
+    if (action === 'update' || action === 'delete') {
         data = _.omitBy(data, _.isUndefined);
-        data = update_account_object(config_root, data);
+        data = await update_account_object(config_root, data);
     }
     return data;
 }
 
 async function update_account_object(config_root, target) {
-    const account_path = config_root + '/accounts';
     let source;
     try {
-        source = await get_config_data(account_path, target.access_keys[0].access_key);
-        target.nsfs_account_config.new_buckets_path = target.nsfs_account_config.new_buckets_path ||
-            source.nsfs_account_config.new_buckets_path;
+        const account_path = target.name ?
+            get_config_file_path(path.join(config_root, accounts_dir_name), target.name) :
+            get_symlink_config_file_path(path.join(config_root, access_keys_dir_name), target.access_keys[0].access_key);
+        source = await get_config_data(account_path);
     } catch (err) {
-        console.error('NSFS Manage command: Could not find account to update');
+        console.error('NSFS Manage command: Could not find account to update', target, err);
         print_account_usage();
     }
     const data = _.merge({}, source, target);
     return data;
 }
 
-async function add_account_config_file(data, accounts_config_path, config_root_backend) {
+async function config_file_exists(fs_context, config_path) {
+    try {
+        await nb_native().fs.stat(fs_context, config_path);
+    } catch (err) {
+        if (err.code === 'ENOENT') return false;
+    }
+    return true;
+}
+
+async function add_account_config_file(data, accounts_path, access_keys_path, config_root_backend) {
     const is_valid = await validate_account_add_args(data);
     if (!is_valid) {
         print_account_usage();
         return;
     }
     // TODO: support non root fs context
     const fs_context = get_root_fs_context(config_root_backend);
-    const full_account_config_path = get_config_file_path(accounts_config_path, data.access_keys[0].access_key);
+    const access_key = data.access_keys[0].access_key;
+    const full_account_config_path = get_config_file_path(accounts_path, data.name);
+    const full_account_config_access_key_path = get_symlink_config_file_path(access_keys_path, access_key);
+
+    const name_exists = await config_file_exists(fs_context, full_account_config_path);
+    const access_key_exists = await config_file_exists(fs_context, full_account_config_access_key_path);
```

Comment:
I think we need the same check for updating the account config file with `new_name` or `new_access_key`.

Reply:
thank you, I added it